src/components/tiles/table/use-content-change-handlers.ts

```diff
@@ -124,22 +125,36 @@ export const useContentChangeHandlers = ({
     if (!readOnly && consumerTile) {
       const sharedModelManager = consumerTile.tileEnv?.sharedModelManager;
       if (sharedModelManager?.isReady) {
+        // If the consumer tile does not support multiple shared data sets, remove it from
+        // any existing shared data set before linking.
+        if (!getTileContentInfo(consumerTile.type)?.supportsMultipleDataSets) {
+          const existingSharedTable = sharedModelManager?.findFirstSharedModelByType(SharedDataSet, tileInfo.id);
```

Comment:
I wonder if we want to start looping through all shared models in cases like this, rather than assuming there's only a single one connected? Theoretically it shouldn't be possible for multiple to be connected, but it wouldn't be too much trouble to just loop through them all.

Reply:
@tealefristoe Thanks. I'm going to leave this as is here, but I'll keep this in mind for the story for linking data cards where I'll be moving this function and refactoring it a bit.